pkg/subctl/cmd/validate_deployment.go

```diff
@@ -0,0 +1,142 @@
+
+	for _, item := range configs {
+		message := fmt.Sprintf("Validating submariner pods in %q", item.clusterName)
+		status.Start(message)
+		fmt.Println()
+		checkPods(item.config, OperatorNamespace)
+	}
+}
+
+func checkPods(config *rest.Config, operatorNamespace string) {
+	submariner := getSubmarinerResource(config)
+	if submariner == nil {
+		status.QueueWarningMessage(submMissingMessage)
+		status.End(cli.Success)
+		return
+	}
+
+	kubeClientSet, err := kubernetes.NewForConfig(config)
+
+	if err != nil {
+		exitOnError("error creating Kubernetes client", err)
+	}
+
+	if !CheckDaemonset(kubeClientSet, operatorNamespace, "submariner-gateway") {
+		return
+	}
+
+	if !CheckDaemonset(kubeClientSet, operatorNamespace, "submariner-routeagent") {
+		return
+	}
+
+	// Check if service-discovery components are deployed and running if enabled
+	if submariner.Spec.ServiceDiscoveryEnabled {
+		// Check lighthouse-agent
+		if !CheckDeployment(kubeClientSet, operatorNamespace, "submariner-lighthouse-agent") {
+			return
+		}
+
+		// Check lighthouse-coreDNS
+		if !CheckDeployment(kubeClientSet, operatorNamespace, "submariner-lighthouse-coredns") {
+			return
+		}
+	}
+	// Check if globalnet components are deployed and running if enabled
+	if submariner.Spec.GlobalCIDR != "" {
+		if !CheckDaemonset(kubeClientSet, operatorNamespace, "submariner-globalnet") {
+			return
+		}
+	}
+
+	message := "All Submariner pods are up and running"
+	status.QueueSuccessMessage(message)
+	status.End(cli.Success)
+}
+
+func CheckDeployment(k8sClient kubernetes.Interface, namespace, deploymentName string) bool {
+	deployment, err := k8sClient.AppsV1().Deployments(namespace).Get(deploymentName, metav1.GetOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining Deployment %q: %v", deploymentName, err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return false
+	}
+
+	var replicas int32 = 1
+	if deployment.Spec.Replicas != nil {
+		replicas = *deployment.Spec.Replicas
+	}
+
+	if deployment.Status.AvailableReplicas != replicas {
+		message := fmt.Sprintf("The desired number of replicas for Deployment %q (%d)"+
+			" does not match the actual number running (%d)", deploymentName, replicas,
+			deployment.Status.AvailableReplicas)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return false
+	}
+
+	return true
+}
+
+func CheckDaemonset(k8sClient kubernetes.Interface, namespace, daemonSetName string) bool {
+	daemonSet, err := k8sClient.AppsV1().DaemonSets(namespace).Get(daemonSetName, metav1.GetOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining Daemonset %q: %v", daemonSetName, err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return false
+	}
+
+	if daemonSet.Status.CurrentNumberScheduled != daemonSet.Status.DesiredNumberScheduled {
+		message := fmt.Sprintf("The desired number of nodes that are running pods for DaemonSet %q (%d)"+
```

Comment:
```suggestion
		message := fmt.Sprintf("The desired number of running pods for DaemonSet %q (%d)"+
```

For some of our DaemonSets like gateway, globalnet etc we use Selectors. Hence they do not run on each and every node of the Cluster. IMO, its better to remote reference to nodes in the message and just say that desired number of running pods does not match with actual number.